controllers/user.js

```diff
@@ -1937,6 +1919,46 @@ async function createRemovalHashWaitlist(req, res) {
   });
 }
 
+async function handleRemovalPilotMgmt(req, res) {
+  if (req.query && Object.entries(req.query).length > 0) {
+    const { email, optout } = req.query;
+    if (email && optout) {
+      let doOptOut;
+      if (optout === "true") {
+        doOptOut = true;
+      } else if (optout === "false") {
+        doOptOut = false;
+      } else {
+        return res.status(400).json({
+          error: "no valid optout status",
+        });
+      }
+      const removeResponse = await DB.handleRemovalOptOutByEmail(
+        email,
+        doOptOut
+      );
+      const removeSuccess = Boolean(removeResponse);
+      if (removeSuccess) {
+        return res.status(200).json({
+          msg: `${email} optout status set to ${doOptOut}`,
+        });
+      } else {
+        return res.status(400).json({
+          error: "no record found",
+        });
+      }
+    } else {
+      return res.status(400).json({
+        error: "no email or optout param",
+      });
+    }
+  } else {
+    return res.status(400).json({
+      error: "no query",
+    });
+  }
+}
```

Comment:
I've started to treat `else` blocks as code "sniffs". I refactor them out by inverting their logic and returning from the function early. It avoid multiple levels of nested logic branches. I also treat `let` as a code "sniff" too. Make immutable `const` vars helps prevent future changes to this code from accidentally changing the variable value in a way that the original code didn't take into consideration.

This should be the same logic, in a more read-able form (not tested):

```suggestion
  if (!req.query || !Object.entries(req.query).length > 0) {
    return res.status(400).json({
      error: "no query",
    });
  }

  const { email, optout } = req.query;
  if (!email || !optout) {
    return res.status(400).json({
      error: "no email or optout param",
    });
  }

  if (!["true", "false"].includes(optout)) {
    return res.status(400).json({
      error: "no valid optout status",
    });
  }

  const doOptOut = (optout === "true");
  const removeResponse = await DB.handleRemovalOptOutByEmail(
    email,
    doOptOut
  );
  const removeSuccess = Boolean(removeResponse);
  if (!removeSuccess) {
    return res.status(400).json({
      error: "no record found",
    });
  }

  return res.status(200).json({
    msg: `${email} optout status set to ${doOptOut}`,
  });
```

Reply:
Thanks! Definitely better structure.